tests/functional/modules/test_zos_job_query_func.py

```diff
@@ -33,3 +33,22 @@ def test_zos_job_query_func(ansible_zos_module):
     for result in results.contacted.values():
         assert result.get("changed") is False
         assert result.get("jobs") is not None
+
+
+# test to show multi wildcard in Job_id query won't crash the search
```

Comment:
Are we able to support job names that start with * , what would that look like if we do for very busy system?

Reply:
Yes... not as different as you might think: regardless of the mask, we ask for something like jls *, then filter the results in the python.  If it's job_id, we get a smaller list, up to the first asterisk; it it's job_name, we get them all.